tests/unit/s2n_cipher_suite_match_test.c

```diff
@@ -781,6 +781,253 @@ int main(int argc, char **argv)
+                .suites = cipher_suites_with_tls12_chacha20,
+                .allow_chacha20_boosting = true
+            };
+
+            const struct s2n_cipher_preferences cipher_preference_with_chacha20_and_no_boosting = {
+                .count = s2n_array_len(cipher_suites_with_chacha20_last),
+                .suites = cipher_suites_with_chacha20_last,
+                .allow_chacha20_boosting = false
+            };
+
+            const struct s2n_cipher_preferences cipher_preference_without_chacha20_and_boosting = {
+                .count = s2n_array_len(cipher_suites_without_chacha20),
+                .suites = cipher_suites_without_chacha20,
+                .allow_chacha20_boosting = true
+            };
+
+            const struct s2n_security_policy test_chacha20_and_boosted = {
+                .minimum_protocol_version = S2N_SSLv3,
+                .cipher_preferences = &cipher_preference_with_chacha20_and_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            const struct s2n_security_policy test_tls12_chacha20_and_boosted = {
+                .minimum_protocol_version = S2N_SSLv2,
+                .cipher_preferences = &cipher_preference_with_tls12_chacha20_and_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            const struct s2n_security_policy test_chacha20_and_not_boosted = {
+                .minimum_protocol_version = S2N_SSLv3,
+                .cipher_preferences = &cipher_preference_with_chacha20_and_no_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            const struct s2n_security_policy test_no_chacha20_and_boosted = {
+                .minimum_protocol_version = S2N_SSLv3,
+                .cipher_preferences = &cipher_preference_without_chacha20_and_boosting,
+                .kem_preferences = &kem_preferences_null,
+                .signature_preferences = &s2n_signature_preferences_20201021,
+                .ecc_preferences = &s2n_ecc_preferences_test_all,
+            };
+
+            uint8_t chacha20_boosted_wire[] = {
+                TLS_CHACHA20_POLY1305_SHA256,
+                TLS_AES_256_GCM_SHA384,
+                TLS_AES_128_GCM_SHA256,
+            };
+
+            EXPECT_SUCCESS(s2n_enable_tls13_in_test());
+            /* If the server chose a security policy w/ ChaCha20 AND has ChaCha20 boosting enabled
+             * then negotiate ChaCha20 iff the client has ChaCha20 as the most-preferred cipher.
+             */
```

Comment:
Try to make these test names shorter and easier to skim through. It can help to make a list of all your test cases before you implement them-- that'll also make it clearer if you're missing any.

I'd also think about if there's a way you can organize / group these tests to help with readability.